### PR TITLE
PLATFORM-2121 - SQL schema update

### PR DESCRIPTION
**Pull request by macbre**
Let's keep sql files in repo in sync with changes introduced by [PLATFORM-2121](https://wikia-inc.atlassian.net/browse/PLATFORM-2121).

@drozdo / @jcellary

### CODE REVIEW

File: extensions/wikia/AchievementsII/schema.sql [hunk 1]
```diff
@@ -69,13 +69,13 @@ CREATE TABLE IF NOT EXISTS `ach_custom_badges` (
...
-	`wiki_id`
-)
-) ENGINE = INNODB CHARACTER SET utf8 COLLATE utf8_unicode_ci;
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `wiki_id` int(9) NOT NULL,
+  `date` datetime NOT NULL,
+  `data` text COLLATE utf8_unicode_ci NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `wiki_id` (`wiki_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```
Comment (drozdo) [inline extensions/wikia/AchievementsII/schema.sql:76]: this primary is not necessary as there is already usable UNIQUE key

File: extensions/wikia/Chat2/sql/chat_bans_users.sql [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `cbu_wiki_id` int(10) NOT NULL DEFAULT '0',
+  `cbu_user_id` int(10) NOT NULL DEFAULT '0',
+  `cbu_admin_user_id` int(10) NOT NULL DEFAULT '0',
+  `reason` varbinary(255) DEFAULT NULL,
+  `start_date` varbinary(14) DEFAULT NULL,
+  `end_date` varbinary(14) DEFAULT NULL,
+  PRIMARY KEY (`id`),
   UNIQUE KEY `cbu_user_id` (`cbu_wiki_id`,`cbu_user_id`),
   KEY `wiki_start_date` (`cbu_wiki_id`,`start_date`)
```
Comment (drozdo) [inline extensions/wikia/Chat2/sql/chat_bans_users.sql:9]: same here

File: extensions/wikia/ContentReview/schema.sql [hunk 3]
```diff
@@ -1,41 +1,92 @@
...
+  `status` smallint(5) unsigned NOT NULL,
+  `submit_user_id` int(10) unsigned NOT NULL,
+  `submit_time` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `review_user_id` int(10) unsigned DEFAULT NULL,
+  `review_start` timestamp NULL DEFAULT NULL,
+  `escalated` tinyint(1) NOT NULL DEFAULT '0',
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `page_id` (`wiki_id`,`page_id`,`revision_id`),
+  UNIQUE KEY `content_review_unique_idx` (`wiki_id`,`page_id`,`status`),
+  KEY `content_review_status_idx` (`status`)
```
Comment (drozdo) [inline extensions/wikia/ContentReview/schema.sql:37]: same here

File: extensions/wikia/ContentReview/schema.sql [hunk 4]
```diff
@@ -1,41 +1,92 @@
...
+  `status` smallint(5) unsigned NOT NULL,
+  `submit_user_id` int(10) unsigned NOT NULL,
+  `submit_time` timestamp NOT NULL DEFAULT '0000-00-00 00:00:00',
+  `review_user_id` int(10) unsigned NOT NULL,
+  `review_start` timestamp NOT NULL DEFAULT '0000-00-00 00:00:00',
+  `review_end` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `escalated` tinyint(1) NOT NULL DEFAULT '0',
+  PRIMARY KEY (`id`),
+  KEY `wiki_id_review_end_idx` (`wiki_id`,`review_end`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```
Comment (drozdo) [inline extensions/wikia/ContentReview/schema.sql:76]: same here

File: maintenance/wikia/sql/dataware-schema.sql [hunk 5]
```diff
@@ -14,562 +14,1046 @@
...
+DROP TABLE IF EXISTS `ab_experiments`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `ab_experiments` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `name` varchar(255) NOT NULL,
+  `description` tinyblob,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `name` (`name`)
+) ENGINE=InnoDB AUTO_INCREMENT=61 DEFAULT CHARSET=latin1;
```
Comment (drozdo) [inline maintenance/wikia/sql/dataware-schema.sql:109]: same here

File: maintenance/wikia/sql/dataware-schema.sql [hunk 6]
```diff
@@ -14,562 +14,1046 @@
...
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `cbu_wiki_id` int(10) NOT NULL DEFAULT '0',
+  `cbu_user_id` int(10) NOT NULL DEFAULT '0',
+  `cbu_admin_user_id` int(10) NOT NULL DEFAULT '0',
+  `reason` varbinary(255) DEFAULT NULL,
+  `start_date` varbinary(14) DEFAULT NULL,
+  `end_date` varbinary(14) DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `cbu_user_id` (`cbu_wiki_id`,`cbu_user_id`),
+  KEY `wiki_start_date` (`cbu_wiki_id`,`start_date`)
```
Comment (drozdo) [inline maintenance/wikia/sql/dataware-schema.sql:164]: same here

File: maintenance/wikia/sql/dataware-schema.sql [hunk 7]
```diff
@@ -14,562 +14,1046 @@
...
+  KEY `user_id` (`user_id`),
+  KEY `stats_idx` (`reviewer_id`,`state`),
+  KEY `priority_index` (`priority`),
+  KEY `last_edited_priority` (`last_edited`,`priority`),
+  KEY `priority_last_edited` (`priority`,`last_edited`),
+  KEY `page_id_idx` (`page_id`),
+  KEY `review_by_edit_prio` (`reviewer_id`,`review_start`,`priority`,`last_edited`),
+  KEY `image_list` (`wiki_id`,`top_200`),
+  KEY `image_list2` (`state`,`top_200`,`last_edited`,`priority`)
 ) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
Comment (drozdo) [inline maintenance/wikia/sql/dataware-schema.sql:301]: 12 fields - 12 keys, this probably needs some work ;)

File: maintenance/wikia/sql/dataware-schema.sql (hunk 5 above)
Comment (macbre) [inline maintenance/wikia/sql/dataware-schema.sql:109]:
```
mysql@master.db-archive.service.consul[dataware]>ALTER TABLE ab_experiments DROP COLUMN id;
ERROR 1829 (HY000): Cannot drop column 'id': needed in a foreign key constraint 'fk_group_experiment' of table 'dataware.ab_experiment_groups'
```

File: maintenance/wikia/sql/dataware-schema.sql [hunk 8]
```diff
@@ -14,562 +14,1046 @@
...
+  `wid` int(11) DEFAULT NULL,
+  `id` int(11) DEFAULT NULL,
+  `title` tinytext,
+  `info_key` tinytext,
+  `value` text,
+  `template` tinytext,
+  `additional` text
 ) ENGINE=InnoDB DEFAULT CHARSET=latin1;
-SET character_set_client = @saved_cs_client;
+/*!40101 SET character_set_client = @saved_cs_client */;
```
Comment (drozdo) [inline maintenance/wikia/sql/dataware-schema.sql:356]: no index here?

File: maintenance/wikia/sql/dataware-schema.sql [hunk 9]
```diff
@@ -14,562 +14,1046 @@
...
-  `mcl_timestamp` int(11) default NULL
+DROP TABLE IF EXISTS `info_schema_map`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `info_schema_map` (
+  `info_key` tinytext,
+  `template` tinytext,
+  `type` tinytext
 ) ENGINE=InnoDB DEFAULT CHARSET=latin1;
-SET character_set_client = @saved_cs_client;
```
Comment (drozdo) [inline maintenance/wikia/sql/dataware-schema.sql:248]: no index here?

File: maintenance/wikia/sql/dataware-schema.sql [hunk 10]
```diff
@@ -14,562 +14,1046 @@
...
+--
+
+DROP TABLE IF EXISTS `info_templates`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `info_templates` (
+  `wid` int(11) DEFAULT NULL,
+  `template` tinytext
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
+/*!40101 SET character_set_client = @saved_cs_client */;
```
Comment (drozdo) [inline maintenance/wikia/sql/dataware-schema.sql:382]: no idnex here?

File: maintenance/wikia/sql/dataware-schema.sql [hunk 11]
```diff
@@ -14,562 +14,1046 @@
...
+  KEY `min_time` (`wiki_id`,`minimum_time`),
+  KEY `max_time` (`wiki_id`,`maximum_time`),
+  KEY `wikitext_size` (`wiki_id`,`wikitext_size`),
+  KEY `html_size` (`wiki_id`,`html_size`),
+  KEY `exp_func_count` (`wiki_id`,`exp_func_count`),
+  KEY `node_count` (`wiki_id`,`node_count`),
+  KEY `post_expand_size` (`wiki_id`,`post_expand_size`),
+  KEY `temp_arg_size` (`wiki_id`,`temp_arg_size`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
+/*!40101 SET character_set_client = @saved_cs_client */;
```
Comment (drozdo) [inline maintenance/wikia/sql/dataware-schema.sql:474]: amount of indexes here is frightening, it should be "analytical" table for some stats, inserting data into it is have to be quite expensive with so many indexes

File: maintenance/wikia/sql/dataware-schema.sql [hunk 12]
```diff
@@ -14,562 +14,1046 @@
...
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `city_id` int(11) NOT NULL,
+  `old_title` varchar(255) NOT NULL,
+  `sanitized_title` varchar(255) NOT NULL,
+  `operation_status` char(14) NOT NULL DEFAULT 'UNKNOWN',
+  `operation_time` char(14) NOT NULL,
+  `article_title` varchar(255) NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `city_id` (`city_id`,`old_title`),
+  KEY `sanitized_title` (`sanitized_title`)
```
Comment (drozdo) [inline maintenance/wikia/sql/dataware-schema.sql:814]: this primary is not necessary as there is already usable UNIQUE key

File: maintenance/wikia/sql/dataware-schema.sql [hunk 13]
```diff
@@ -14,562 +14,1046 @@
...
+CREATE TABLE `video_postmigrate_undo` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `wiki_id` int(11) NOT NULL,
+  `entry_table` varchar(255) NOT NULL,
+  `entry_id` varchar(255) NOT NULL,
+  `entry_id_field` varchar(255) NOT NULL,
+  `entry_ns_field` varchar(255) NOT NULL,
+  PRIMARY KEY (`id`),
+  KEY `vpu_wiki_id` (`wiki_id`)
+) ENGINE=InnoDB AUTO_INCREMENT=798955 DEFAULT CHARSET=latin1;
```
Comment (drozdo) [inline maintenance/wikia/sql/dataware-schema.sql:872]: primary key on wiki_id+entry_id should be fine?

File: maintenance/wikia/sql/dataware-schema.sql (hunk 8 above)
Comment (macbre) [inline maintenance/wikia/sql/dataware-schema.sql:356]: `info_*` tables where created in March 2016. Let's find the owner.

File: maintenance/wikia/sql/stats-schema.sql [hunk 14]
```diff
@@ -441,9 +360,37 @@ CREATE TABLE `tags_stats_filter` (
   `sf_type` enum('blog','user','article','city') DEFAULT NULL,
   PRIMARY KEY (`sf_id`),
   UNIQUE KEY `sf_city_id` (`sf_city_id`,`sf_page_id`,`sf_user_id`,`sf_tag_id`,`sf_type`)
+) ENGINE=InnoDB AUTO_INCREMENT=645 DEFAULT CHARSET=latin1;
```
Comment (drozdo) [inline maintenance/wikia/sql/stats-schema.sql]: this primary is not necessary as there is already usable UNIQUE key

File: maintenance/wikia/sql/dataware-schema.sql (hunk 11 above)
Comment (macbre) [inline maintenance/wikia/sql/dataware-schema.sql:474]: Pinging @wladekb here as the author of `ParserSpeed`

File: maintenance/wikia/sql/dataware-schema.sql (hunk 5 above)
Comment (macbre) [inline maintenance/wikia/sql/dataware-schema.sql]:
```sql
ALTER TABLE ab_experiments DROP PRIMARY KEY;
```

File: maintenance/wikia/sql/dataware-schema.sql (hunk 12 above)
Comment (macbre) [inline maintenance/wikia/sql/dataware-schema.sql:814]: @garthwebb: do we need this table? It look like a leftover after videos migration.

File: maintenance/wikia/sql/dataware-schema.sql (hunk 13 above)
Comment (macbre) [inline maintenance/wikia/sql/dataware-schema.sql:872]: @garthwebb: do we need this table? It look like a leftover after videos migration.

File: maintenance/wikia/sql/stats-schema.sql (hunk 14 above)
Comment (macbre) [inline maintenance/wikia/sql/stats-schema.sql]: `tags_stats_filter` does not seem to be used, I'll rename it.

File: maintenance/wikia/sql/dataware-schema.sql (hunk 7 above)
Comment (macbre) [inline maintenance/wikia/sql/dataware-schema.sql:301]: Not really sure who owns image_review now.

Comment (macbre) [inline maintenance/wikia/sql/dataware-schema.sql:301]: Actually, since today it's SUS -> [MAIN-6918](https://wikia-inc.atlassian.net/browse/MAIN-6918)

**Pull request merged**